Improve Contact Form

jonatanklosko opened the pull request:
Fixes #2690. You can try it out [here](https://staging.worldcubeassociation.org/contact/website) and expect emails to be visible [here](https://staging.worldcubeassociation.org:444) (may require refreshing the mailbox).

jonatanklosko commented on WcaOnRails/app/views/contacts/website.html.erb:
```diff
@@ -5,6 +5,15 @@
+      [t('.inquiries.competitions_in_general'), "wct"],
+      [t('.inquiries.wca_id_or_profile'), "wrt"],
+      [t('.inquiries.media'), "wrt"],
+      [t('.inquiries.other'), "wct"],
```
@UnsolvedCypher should WCT get all of those inquiries or should some be WRT-only? Please let me know where should each of these go =)

Laura-O commented:
It's difficult to select the radio buttons on mobile screens as they do not have a margin on the left side:
![bildschirmfoto 2018-06-15 um 13 43 57](https://user-images.githubusercontent.com/362249/41466371-47523c0a-70a2-11e8-85ef-28645828f712.png)

jonatanklosko commented:
You can select them by clicking on corresponding text, but the whole form should have margin indeed! Good catch =)

Btw. any idea for new introduction text above the form?

jonatanklosko commented on WcaOnRails/app/models/website_contact.rb:
```diff
@@ -2,4 +2,16 @@
+  attr_accessor :inquiry
+  attr_accessor :competition_id
+
+  # Override the `to_mail` validation, to show errors for `inquiry` instead.
```
Well, it may fail in two cases:
- there is no `to_email` at all (this is the case if someone doesn't specify `inquiry` in this case)
- we have a typo in our code that sets it to an invalid email

We could certainly have presence validation on `inquiry` instead and drop the validation on `to_email` (for all contact forms), assuming that we ensure it's validity on our own. The `to_email` was here, that's why I left it ^^ Thoughts?

jfly replied:
Ah, gotcha. I think part of my confusion is coming from the fact that the mapping between inquiry to destination email address is in the controller, instead of over here. Have you considered maybe moving that mapping into this `WebsiteContact` model? I think that could make it clearer what's going on. I think I'd implement this by either implementing our own `def to_email` method, or if that proves tricky perhaps by explicitly setting `to_email` in a before validation block.

jonatanklosko replied:
I thought about `before_validation` callback, but `WebsiteContact` is not an `ActiveRecord` (just includes some validations), so it doesn't support all those neat hooks. Overriding `to_email` sounds like a plan though, gonna give it a shot!

jonatanklosko commented:
I removed the inaccurate description entirely, there is always the "Other' option that would allow WCT decide whether to redirect to another team or not. If WCT spots another common inquiry that we don't have listed, we can easily add it =) Also, I added a "Software" inquiry that gets sent straight to WST!
